modules/router-store/src/router_selectors.ts

```diff
@@ -2,52 +2,60 @@ import { createSelector } from '@ngrx/store';
 import { RouterStateSelectors } from './models';
 import { RouterReducerState } from './reducer';
 
-export function getSelectors<V>(
-  selectState: (state: V) => RouterReducerState<any>
-): RouterStateSelectors<V>;
 export function getSelectors<V>(
   selectState: (state: V) => RouterReducerState<any>
 ): RouterStateSelectors<V> {
   const selectRouterState = createSelector(
     selectState,
-    (router) => router && router.state
+    (router) => router?.state
   );
-  const selectCurrentRoute = createSelector(
+  const selectRootRoute = createSelector(
     selectRouterState,
-    (routerState) => {
-      if (!routerState) {
-        return undefined;
-      }
-      let route = routerState.root;
-      while (route.firstChild) {
-        route = route.firstChild;
-      }
-      return route;
-    }
+    (routerState) => routerState?.root
   );
+  const selectCurrentRoute = createSelector(selectRootRoute, (rootRoute) => {
+    let route = rootRoute;
+    while (route?.firstChild) {
+      route = route.firstChild;
+    }
+    return route;
+  });
   const selectFragment = createSelector(
-    selectCurrentRoute,
-    (route) => route && route.fragment
+    selectRootRoute,
+    (route) => route?.fragment
   );
   const selectQueryParams = createSelector(
-    selectCurrentRoute,
-    (route) => route && route.queryParams
+    selectRootRoute,
+    (route) => route?.queryParams
   );
   const selectQueryParam = (param: string) =>
-    createSelector(selectQueryParams, (params) => params && params[param]);
+    createSelector(selectQueryParams, (params) => params?.[param]);
   const selectRouteParams = createSelector(
     selectCurrentRoute,
-    (route) => route && route.params
+    (route) => route?.params
   );
   const selectRouteParam = (param: string) =>
```

Comment:
This shouldn't modify the existing selector, but add a new selector, such as `selectParamFromRouteTree` or something similar, so there's a difference between selecting the current one, and the first occurrence of the parameter.

I think this should be in a separate PR from the perf changes also

Reply:
Okay, I'll create a separate PR for new selector.